docs/spec/concepts.rst

```diff
@@ -3,6 +3,268 @@
+
+A gradual type ``A`` is consistent with a gradual type ``B``, and ``B`` is
+consistent with ``A``, if and only if there exists some fully static type ``C``
+which is a materialization of both ``A`` and ``B``.
+
+``Any`` is consistent with every type, and every type is consistent with
+``Any``. (This follows from the definitions of materialization and consistency
+but is worth stating explicitly.)
+
+The consistency relation is not transitive. ``tuple[int, int]`` is consistent
+with ``tuple[Any, int]``, and ``tuple[Any, int]`` is consistent with
+``tuple[str, int]``, but ``tuple[int, int]`` is not consistent with
+``tuple[str, int]``.
+
+The consistency relation is symmetric. If ``A`` is consistent with ``B``, ``B``
+is also consistent with ``A``. It is also reflexive: ``A`` is always consistent
+with ``A``.
+
+The assignable-to (or consistent subtyping) relation
+----------------------------------------------------
+
+Given the materialization relation and the subtyping relation, we can define
+the **consistent subtype** relation over all types. A type ``B`` is a
+consistent subtype of a type ``A`` if there exists a materialization ``A'`` of
+``A`` and a materialization ``B'`` of ``B``, where ``A'`` and ``B'`` are both
+fully static types, and ``B'`` is a subtype of ``A'``.
+
+Consistent subtyping defines "assignability" for Python.  An expression can be
+assigned to a variable (including passed as a parameter or returned from a
+function) if it is a consistent subtype of the variable's type annotation
+(respectively, parameter's type annotation or return type annotation).
+
+We can say that a type ``B`` is "assignable to" a type ``A`` if ``B`` is a
+consistent subtype of ``A``.
+
+In the remainder of this specification, we will usually prefer the term
+**assignable to** over "consistent subtype of". The two are synonymous, but
+"assignable to" is shorter, and may communicate a clearer intuition to many
+readers.
+
+For example, ``Any`` is assignable to ``int``, because ``int`` is a
+materialization of ``Any``, and ``int`` is a subtype of ``int``. The same
+materialization also gives that ``int`` is assignable to ``Any``.
+
+The assignable-to relation is not generally symmetric, though. If ``B`` is a
+subtype of ``A``, then ``tuple[Any, B]`` is assignable to ``tuple[int, A]``,
+because ``tuple[Any, B]`` can materialize to ``tuple[int, B]``, which is a
+subtype of ``tuple[int, A]``. But ``tuple[int, A]`` is not assignable to
+``tuple[Any, B]``.
+
+Attributes and methods
+----------------------
+
+In Python, we can do more with objects at runtime than just assign them to
+names, pass them to functions, or return them from functions. We can also
+get/set attributes and call methods.
+
+In the Python object model, the operations that can be performed on a value all
+de-sugar to method calls. For example, ``a + b`` is syntactic sugar for either
+``a.__add__(b)`` or ``b.__radd__(a)``.
```

Comment:
We may as well be precise here, since this is a specification, after all. Special methods are always looked up on the class :P

```suggestion
``type(a).__add__(a, b)`` or ``type(b).__radd__(b, a)``.
```

Reply:
This is also not pedantically correct, though:

```
class C(type):
    def __getattribute__(self, x):
        return lambda x, y: ("meta", x, y)

class A(metaclass=C):
    def __add__(self, other):
        return ("method", self, other)
    
    def __repr__(self) -> str:
        return "A()"

a = A()
b = A()
print(a + b)  # ('method', A(), A())
print(type(a).__add__(a, b))  # ('meta', A(), A())
```

Reply:
I guess. Maybe we should just hedge slightly and say that it "approximately" desugars to ``a.__add__(b)`` or ``b.__radd__(a)``, and link to https://snarky.ca/unravelling-binary-arithmetic-operations-in-python/

Reply:
Ultimately I think it would be good for the typing spec to take a clear stand on whether overriding `__getattribute__` is or isn't part of the type system's (abstracted) model of the Python object model. Personally, I'm not sure it should be.

But this PR isn't the place to make that call. I'll adjust this wording to be accurate without getting too far into the weeds.